File: io-hotmoka-service/src/main/java/io/hotmoka/service/models/responses/MethodCallTransactionResponseModel.java
package io.hotmoka.service.models.responses;

import io.hotmoka.beans.responses.MethodCallTransactionResponse;

public abstract class MethodCallTransactionResponseModel extends CodeExecutionTransactionResponseModel {

	/**
	 * True if and only if the call was charged to the receiver of the target method
	 * rather than to the caller of the transaction.
	 */
	public boolean selfCharged;

	protected MethodCallTransactionResponseModel(MethodCallTransactionResponse response) {
        super(response);

        this.selfCharged = response.selfCharged;
	}

	protected MethodCallTransactionResponseModel() {}
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-js/src/models/responses/MethodCallTransactionResponseModel.ts
import {CodeExecutionTransactionResponseModel} from "./CodeExecutionTransactionResponseModel";
import {UpdateModel} from "../updates/UpdateModel";

export abstract class MethodCallTransactionResponseModel extends CodeExecutionTransactionResponseModel {
    /**
     * True if and only if the call was charged to the receiver of the target method
     * rather than to the caller of the transaction.
     */
    selfCharged: boolean

    protected constructor(
        updates: Array<UpdateModel>,
        gasConsumedForCPU: string,
        gasConsumedForRAM: string,
        gasConsumedForStorage: string,
        selfCharged: boolean
    ) {
        super(updates, gasConsumedForCPU, gasConsumedForRAM, gasConsumedForStorage)
        this.selfCharged = selfCharged
    }
}